### DISCUSSION TITLE
Where can I find more information about "hints" appearing next to disassembled instructions?

### DISCUSSION THREAD

**Discussion by tosroka** [radareorg/radare2#21601] [Q&A] [answered]
The hints/comments I'm talking about is everything that starts with `;`, for example:
```asm
call sym.imp.setlocale      ; char *setlocale(int category, const char *locale)                                                                    
lea rsi, str._usr_share_locale ;[7] ;[8] ; 0x19c79 ; "/usr/share/locale" ; char *dirnam
```
I see it shows a bunch of useful information, from visual mode help I was able to learn that `r` cycles between different hint types and `[1-9]` can be used to follow the "shortcuts" connected to corresponding numbers (while writing this I accidentally noticed they are different in graph view, [o?] things).

I can't find anywhere a proper explanation about these hints, where should I look?

**Comment by trufae** [comment, marked as answer]
Those are automatic comments, the ;[X] are keyboard shortcuts to jump into the referenced address, yes the 'r' cycles between jumps/coderefs/immediates/..

in visual its [o????] because in disasm only supports 10 keyboard but the graph can contain zillions of nodes and you cant address them with just 1 number so a letter is required.

I dont think there's anything extra to explain, maybe in the book there are some lines about it, but i think those comments are pretty clear. they reference the actual string , function signature with typed arguments or numeric address that the given instruction reference.

The disasm loop is also showing many more autocomments like reloc information, esil computations, character mapping, imports, user comments, etc
